aten/src/ATen/templates/TensorBody.h

```diff
@@ -48,6 +48,7 @@ namespace at {
 
 class Tensor;
 using TensorList = ArrayRef<Tensor>;
+using ScalarList = ArrayRef<Scalar>;
```

Comment:
This is a minor point (and implies some divergence from TensorList), but I'd prefer it if we stopped defining so many aliases for these types. ScalarList is doubly bad because it implies an owning list (which an ArrayRef is not). You don't need the typedef unless you actually translate `Scalar[]` into `ScalarList`, and preventing this translation is as simple as removing a case from the codegen...